pkg/apis/common/v1/config.go

```diff
@@ -27,7 +27,7 @@ func NewConfig(cfg map[string]interface{}) Config {
 }
 
 // MarshalJSON implements the Marshaler interface.
-func (c *Config) MarshalJSON() ([]byte, error) {
+func (c Config) MarshalJSON() ([]byte, error) {
```

Comment:
No, in the case of `UnmarshalJSON` we absolutely want to only use pointers to mutate the caller to get the result. We don't need this when marshalling (`MarshalJSON()`) because the result is returned.

```go
result, err := json.Marshal(configPointerOrVariable)
err := json.Unmarshal(bytes, configPointer)
```